packages/suite/src/components/suite/BioAuthGuard/BioAuthGuard.tsx

```diff
@@ -152,11 +128,28 @@ export const BioAuthGuard = ({ children }: { children: React.ReactNode }) => {
         return () => {
             window.removeEventListener('blur', handleBlur);
             window.removeEventListener('focus', handleFocus);
+            clearTimeout(timeoutId);
         };
-    }, [dispatch, isBioAuthAvailable]);
+    }, [dispatch, isBioAuthAvailable, isBioAuthEnabled]);
 
-    return isAppUiHidden || isBioAuthValidationRequired ? (
-        <BioAuthOverlay isBioAuthValidationRequired={isBioAuthValidationRequired} />
+    useEffect(() => {
```

Comment:
This is not great, there is a risk of the desktop api being called over and over when this is called from `useEffect` it could crash the electron process.

There should be at least some check that the call is in progress.